exporter/datadogexporter/internal/logs/sender_test.go

```diff
@@ -0,0 +1,212 @@
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 1",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 1",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+			}},
+			testFn: func(jsonLogs testutil.JSONLogs, counter int) {
+				switch counter {
+				case 0:
+					assert.True(t, jsonLogs.HasDDTag("tag1:true"))
+					assert.Len(t, jsonLogs, 1)
+				case 1:
+					assert.True(t, jsonLogs.HasDDTag("tag2:true"))
+					assert.Len(t, jsonLogs, 1)
+				default:
+					t.Fail()
+				}
+			},
+			numRequests: 2,
+		},
+		{
+			name: "two-batches",
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 1",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 1",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag1:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 2",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag1:true",
+					"hostname": "hostname",
+					"message":  "log 2",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 3",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 3",
+					"service":  "server",
+				},
+			}, {
+				Ddsource: datadog.PtrString("golang"),
+				Ddtags:   datadog.PtrString("tag2:true"),
+				Hostname: datadog.PtrString("hostname"),
+				Message:  "log 4",
+				Service:  datadog.PtrString("server"),
+				UnparsedObject: map[string]interface{}{
+					"ddsource": "golang",
+					"ddtags":   "tag2:true",
+					"hostname": "hostname",
+					"message":  "log 4",
+					"service":  "server",
+				},
+			}},
+			testFn: func(jsonLogs testutil.JSONLogs, counter int) {
```

Comment:
```suggestion
			testFn: func(jsonLogs testutil.JSONLogs, call int) {
```
I would call this `call` throughout, to make it obvious what it is. `counter` is a bit ambiguous.